lib/rouge/lexers/ecl.rb

```diff
@@ -0,0 +1,172 @@
+# -*- codding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class ECL < RegexLexer
+      tag 'ecl'
+      filenames '*.ecl'
+      mimetypes 'application/x-ecl'
+
+      title "ECL"
+      desc "Enterprise Control Language (hpccsystems.com)"
+
+      id = /(#?)\b([a-z_][\w]*?)(\d*)\b/i
+
+     class1 = %w(
+        file date str math metaphone metaphone3 uni audit blas system
+      )
+
+      class2 = %w(
+        debug email job log thorlib util workunit
+      )
```

Comment:
@dehilsterlexis Why did you revert these to be local variables?

Reply:
For the root rule which uses them:
`rule %r(\b(?i:(std)\.(#{class1.join('|')})\.(#{class2.join('|')}))\b), Name::Class`

Reply:
Right but for the reasons explained previously, we try to avoid interpolations in regular expressions. The code I wrote avoided the need to do this. I don't understand what the problem with it was.

Reply:
@pyrmont I do. But my previous solution was to remove it. Is there another way to do the pattern matching not using variables?

Reply:
I'm not sure I follow your comment. The code you've changed it to uses both local variables and regex interpolation. I'm not sure what was broken about my approach.

Reply:
@pyrmont I will try it with regex interp.

Reply:
We may be talking cross-purposes. We want to _avoid_ regex interpolation. The code I wrote below does that:

```ruby
rule %r(\b(\w+)\.(\w+)\.(\w+)) do |m|
  if m[1] == "std" &&
    self.class.class_first.include?(m[2]) &&
    self.class.class_second.include?(m[3])
    token Name::Class
  else
    token Name::Variable
  end
end
```

Reply:
(That code is still in the lexer, starting at line 20. Incidentally, it will break now because the `self.class.class_first` and self.class.class_second` methods have been removed.)

Reply:
@pyrmont I did not see you had put in the code already. My bad. I will put it back.

Reply:
@pyrmont restored.